src/commands.ts

```diff
@@ -161,8 +162,25 @@ async function connectCMD(shell:Shell, args: string[]) {
     const gate: Gate = shell.getGate(hostname)
     if (!gate)
         return shell.t.writeln(`Host not found: ${hostname}`)
-    await new Promise<void>((resolve) => {
+    await new Promise<void>(async (resolve) => {
+        gate.onFailure = reason => {
+            terminal7.log(`Connect command got failure ${reason}`) 
+            gate.close()
+            terminal7.storeGates()
+            resolve()
+        }
+        if (Capacitor.isNativePlatform())  {
+            if (!gate.username) {
+                try {
+                    gate.username = await shell.askValue("Username")
+                } catch (e) {
+                    gate.notify("Failed to get username")
+                }
+            }
+        }
+        shell.startWatchdog().catch(e => gate.handleFailure(e))
```

Comment:
It's problematic because if I change it to `gate.handleFailure` then I need to add `.bind(gate)` which seems redundant so I'd rather have it this way.